state_machine/applications/flight/lib/IGRF.py

```diff
@@ -0,0 +1,219 @@
+# flake8: noqa E741
+# Based on: https://github.com/RoboticExplorationLab/pycubed_circuitpython/blob/master/IGRF/igrf.py
+# Fifth order approximatin.
+
+try:
+    import ulab as np
+except ImportError:
+    import numpy as np
+import math
+
+def reset_array(input_array):
+    for i in range(len(input_array)):
+        input_array[i] = 0.0
+
+def unix_time_to_years_since_2020(unix_time):
+    twentytwenty = 1577854800  # 2022-01-01 00:00:00 time stamp
+    t = (unix_time - twentytwenty) / 31557600  # seconds in a Julian astronomical year
+    return t
+
+def igrf13_5(gh, date, latitude_degrees, elongitude_degrees, r_norm_km, cl, sl, p, q):
```

Comment:
i would add a comment somewhere that tells us what r_norm_km, cl, sl, p, and q stand for

Reply:
honestly igrf13_5 is just copy pasted from the port and modified to use unix time. I think it's best to just leave the internals alone even though their a mess. I think the interface functions are pretty well documented. 

I've added an _ to the function name to indicate that it's internal and not meant to be used.